src/data/finset/lattice.lean

```diff
@@ -37,6 +37,14 @@ fold_cons h
 @[simp] lemma sup_insert [decidable_eq β] {b : β} : (insert b s : finset β).sup f = f b ⊔ s.sup f :=
 fold_insert_idem
 
+lemma sup_image [decidable_eq β] (s : finset γ) (f : γ → β) (g : β → α):
+  (s.image f).sup g = s.sup (g ∘ f) :=
+fold_image_idem
+
+@[simp] lemma sup_map (s : finset γ) (f : γ ↪ β) (g : β → α) :
+  (s.map f).sup g = s.sup (g ∘ f) :=
+by simp [finset.sup]
```

Comment:
Out of curiosity, does `fold_map` work as the proof for this?

Reply:
Almost certainly, I think that's what this simp ends up finding

Reply:
Sure - but I mean can we just write `fold_map` as the term-mode proof, rather than asking simp to unfold `sup` then go looking for `fold_map`.